src/map/index.es

```diff
@@ -43,6 +40,7 @@ class Map extends PureComponent {
 
   setMap(map) {
     this.map = map;
+    this.staticContext = this.getDefaultContext();
```

Comment:
this only works because you call `setState` right after. Best to move this object in the constructor after you bind methods and mutate map prop in it. it doesn't participate in react render cycle, so it doesn't matter if props change.

btw you could copy `bindTo` util here to simplify the code a bit. check if it is needed anywhere else.

Reply:
bind only needs here for 2 methods.